### PR TITLE
Added login, styled using bootstrap

### PR DESCRIPTION
**Pull request by ghost**
Sorry about the big pull, I styled using some bootstrap.  Next up is adding a navbar with a logout button and sparkies logo.

### CODE REVIEW

**Comment by Roughsketch** [comment]
I don't know if you're still working on this, but just a heads up that I can't really review it until you fix the merge conflicts.

File: static/js/main.js [hunk 1]
```diff
@@ -128,43 +128,45 @@ function goBack() {
...
+            }else{
+              $( '#errorcon' ).text( 'Error: ' + jsonobj.error );
+            }
         }
-    };
-    xhttp.open('POST', '/api/login', true);
-    xhttp.send(loginfo);
-    return jsonobj;
+    }
+    xhttp.open('POST','api/login', true);
```
Comment (Roughsketch) [inline static/js/main.js]: Paths relative to `/` should start with `/`. Change the link back.

File: static/js/main.js [hunk 2]
```diff
@@ -128,43 +128,45 @@ function goBack() {
...
-        if (xhttp.readyState == XMLHttpRequest.DONE && xhttp.status == 200) {
+    xhttp.onreadystatechange = function(){
+        if(xhttp.readyState == XMLHttpRequest.DONE && xhttp.status == 200){
             jsonobj = JSON.parse(xhttp.responseText);
             console.log(jsonobj);
         }
-    };
-    xhttp.open('GET', '/api/list', true);
+    }
+    xhttp.open('GET', 'api/list', true);
```
Comment (Roughsketch) [inline static/js/main.js]: Paths relative to `/` should start with `/`. Change the link back.

File: static/js/main.js [hunk 3]
```diff
@@ -128,43 +128,45 @@ function goBack() {
...
 }
 
-function login(loginfo) {
+function login(){
     var jsonobj;
+    var user = $('#user').val();
+    var password = $('#password').val();
+    var loginfo = {"user" : "user", "pass" : "pass"};
+    loginfo.user = user;
+    loginfo.pass = password;
```
Comment (Roughsketch) [inline static/js/main.js]: You don't need to set temp values and then immediately overwrite them. Just do 
```js
var loginfo = {"user": user, "pass": password};
```

**Review by Roughsketch — CHANGES_REQUESTED** [review]
The style in this project is inconsistent. I tried to fix it before, but you seem to have undone only some of the changes. I'd recommend finding a JS linter and a style guideline. You can start here: https://eslint.org/.

Style isn't necessary to get this PR approved, but it's something to consider if you plan on working on it more.

**Comment by ghost** [comment]
Got it, should be done just waiting for testing.  I'll leave the temporary data add loop for now unless it really bugs you.  Seems the server is locking the database tho and not allowing new entries.

File: static/js/main.js [hunk 4]
```diff
@@ -55,15 +59,14 @@ function refresh() {
...
-    newxBeeArray = getNodes();
     var alength = newxBeeArray.length;
     var length = xBeeArray.length;
     var found = false;
     var xBeeOBJ = '';
     for (var i = 0; i < length; i++) {
         xBeeArray[i].ConnStatus = false;
     }
-    for (var i = 0; i < alength; i++) {
+    for (i = 0; i < alength; i++) {
```
Comment (Roughsketch) [inline static/js/main.js:69]: Use var, otherwise `i` goes into global scope.

File: static/js/main.js [hunk 5]
```diff
@@ -80,7 +83,7 @@ function refresh() {
     }
     alength = newxBeeArray.length;
     if (alength > 0) {
-        for (var i = 0; i < alength; i++) {
+        for (i = 0; i < alength; i++) {
```
Comment (Roughsketch) [inline static/js/main.js:86]: Same

File: static/js/main.js [hunk 6]
```diff
@@ -128,43 +131,86 @@ function goBack() {
...
+                window.location = 'login.html';
+            }
+        }
+    };
+    xhttp.open('GET','/api/logout', true);
+    xhttp.send();
 }
 
+function addXbees(i){
+    jsonobj = '';
```
Comment (Roughsketch) [inline static/js/main.js:200]: `jsonobj` is not used in the function, just inside the closure. Remove this one and add `var` to the other.

File: static/js/main.js [hunk 7]
```diff
@@ -128,43 +131,86 @@ function goBack() {
...
+    var xhttp = new XMLHttpRequest();
+
+    xbee = {"node_id":i, "name":i.toString(), "units":i.toString()};
```
Comment (Roughsketch) [inline static/js/main.js:203]: Same, but `xbee`. Also not sure why the test data uses the same information for all fields.

File: static/js/main.js [hunk 8]
```diff
@@ -128,43 +131,86 @@ function goBack() {
...
 
-login({
-    "user": "root",
-    "pass": "toor"
-});
-getNodes();
\ No newline at end of file
+    xhttp.onreadystatechange = function(){
+        if(xhttp.readState == XMLHttpRequest.DONE && xhttp.status == 200){
+            jsonobj = JSON.parse(xhttp.responseText);
```
Comment (Roughsketch) [inline static/js/main.js:207]: Same, but `jsonobj`

File: static/js/main.js [hunk 9]
```diff
@@ -128,43 +131,86 @@ function goBack() {
...
+    xhttp.setRequestHeader("Content-Type", "application/json");
+    xhttp.send(JSON.stringify(loginfo));
+}
+
+function logout(){
+    var jsonobj = '';
+    var xhttp = new XMLHttpRequest();
+    xhttp.onreadystatechange = function(){
+        if(xhttp.readyState == XMLHttpRequest.DONE && xhttp.status == 200){
+            jsonobj = JSON.parse(xhttp.responseText);
```
Comment (Roughsketch) [inline static/js/main.js:189]: Remove the outer `jsonobj` and add var to the inner one. `login` and `getNodes` also have the same problem.

**Review by Roughsketch — CHANGES_REQUESTED** [review]
If you let me know what the panic says I can help debug it.

**Comment by Roughsketch** [comment]
Nevermind, I'll just do it.

File: static/js/main.js (hunk 4 above)
Comment (ghost) [reply static/js/main.js:69]: Wierd, wonder if that was my linter...  I'll make sure it's not doing that anymore.

**Pull request merged**